lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -100,8 +105,17 @@ export class PublicClientApplication {
      * or the library, depending on the origin of the error, or the AuthResponse object 
      * containing data from the server (returned with a null or non-blocking error).
      */
-    handleRedirectCallback(authCallback: AuthCallback): void {
-        throw new Error("Method not implemented.");
+    async handleRedirectCallback(authCallback: AuthCallback): Promise<void> {
+        if(!authCallback) {
+            throw BrowserConfigurationAuthError.createInvalidCallbackObjectError(authCallback);
+        }
+
+        this.authCallback = authCallback;
+        const { location: { hash }} = window;
+        if (UrlString.hashContainsKnownProperties(hash)) {
+            this.interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.authCallback);
```

Comment:
Is there a benefit to setting the instance var here? Looks like it is not used again

Reply:
Not sure what you mean by instance var, can you elaborate? 

For some context: This function gets called when we come back from a redirect, so we don't have any context until we understand that the hash has properties we need to handle.

Reply:
why do we set this.interactionHandler on PCA ?

Reply:
I set it rather than passing around a handle to separate functions. In the redirect scenario, a new one is created every time, but in the popup scenario, we would probably just end up using the same instance for every call, so I made it a field of the PCA class. Open to suggestions if we feel there is a better way to handle this.